### PR TITLE
Add hashlib-like Checksum helper class.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: src/crc32c/__init__.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
 # NOTE: ``__config__`` **must** be the first import because it (may)
 #       modify the search path used to locate shared libraries.
+import struct
```
Comment (dhermes) [inline src/crc32c/__init__.py:13]: @tseaver Can you move this **above** the comment about importing `__config__`?

File: src/crc32c/__init__.py [hunk 2]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
         int: New CRC checksum computed for ``chunk``.
     """
     return crc32c._crc32c_cffi.lib.crc32c_value(chunk, len(chunk))
+
+
+class Checksum(object):
+    """Hashlib-alike helper for CRC32C operations."""
+    __slots__ = ('_crc',)
```
Comment (dhermes) [inline src/crc32c/__init__.py]: Can you run `black --line-length=79 $(git ls-files -- '*.py')`?

File: src/crc32c/__init__.py [hunk 3]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
...
+
+    def __init__(self, initial_value=b''):
+        """Construct the helper.
+
+        Args:
```
Comment (dhermes) [inline src/crc32c/__init__.py]: Typically we don't put a docstring on `__init__` and put stuff like this in the class docstring

File: src/crc32c/__init__.py [hunk 4]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
...
+            initial_value (Optional[bytes]): the initial chunk of data from
+            which the CRC32C checksum is computed.  Defaults to b''.
```
Comment (dhermes) [inline src/crc32c/__init__.py:54]:
```suggestion
                which the CRC32C checksum is computed.  Defaults to b''.
```

File: src/crc32c/__init__.py [hunk 5]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
...
+        """
+        self._crc = value(initial_value)
+
+    def __int__(self):
```
Comment (dhermes) [inline src/crc32c/__init__.py]: Why is this needed?

File: src/crc32c/__init__.py [hunk 6]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
...
+        return self._crc
+
+    def update(self, chunk):
+        """Update the checksum with a new chunk of data.
+
+        Args:
+            chunk (Optional[bytes]): a chunk of data used to extend
+            the CRC32C checksum.
```
Comment (dhermes) [inline src/crc32c/__init__.py]:
```suggestion
                the CRC32C checksum.
```

File: src/crc32c/__init__.py [hunk 7]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
...
+
+        Per: https://cloud.google.com/storage/docs/json_api/v1/objects#crc32c
+
+        Returns:
+            bytes: An eight-byte digest string.
+        """
+        return struct.pack('>L', self._crc)
+
+    def hexdigest(self):
+        """Like digest() except returned as a bytestring of double length.
```
Comment (dhermes) [inline src/crc32c/__init__.py]:
```suggestion
        """Like :meth:`digest` except returned as a bytestring of double length.
```

File: src/crc32c/__init__.py [hunk 8]
```diff
@@ -42,3 +44,57 @@ def value(chunk):
...
+        Args:
+            chunk (Optional[bytes]): a chunk of data used to extend
+            the CRC32C checksum.
+        """
+        self._crc = extend(self._crc, chunk)
+
+    def digest(self):
+        """Big-endian order, per RFC 4960.
+
+        Per: https://cloud.google.com/storage/docs/json_api/v1/objects#crc32c
```
Comment (dhermes) [inline src/crc32c/__init__.py]: This is vague. What is "Per:" in reference to?

File: src/crc32c/__init__.py (hunk 2 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: I can live with some of its choices (`"`, meh), but reformatting the table from the RFC?  What is the comment syntax to bail out of that?

Comment (dhermes) [reply src/crc32c/__init__.py]: There is no bail-out for `black`. It's purpose is to avoid such arguments.

File: src/crc32c/__init__.py (hunk 5 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: I'm using it for test assertions.  I can use the private variable instead, if you prefer.

File: src/crc32c/__init__.py (hunk 1 above)
Comment (tseaver) [reply src/crc32c/__init__.py:13]: 7b9fa27

File: src/crc32c/__init__.py (hunk 2 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: d6d73fd adds the example comments from the RFC.  6361bb5 blackens the two files.

File: src/crc32c/__init__.py (hunk 3 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: 7b9fa27

File: src/crc32c/__init__.py (hunk 4 above)
Comment (tseaver) [reply src/crc32c/__init__.py:54]: 7b9fa27

File: src/crc32c/__init__.py (hunk 6 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: 7b9fa27

File: src/crc32c/__init__.py (hunk 7 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: 7b9fa27

File: src/crc32c/__init__.py (hunk 8 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: 7b9fa27

File: src/crc32c/__init__.py (hunk 5 above)
Comment (dhermes) [reply src/crc32c/__init__.py]: I'd prefer, just to have a smaller code footprint here.

**Review by dhermes — APPROVED** [review]
LGTM

File: src/crc32c/__init__.py (hunk 5 above)
Comment (tseaver) [reply src/crc32c/__init__.py]: 89a8520
